test/js/calc-length.js

```diff
@@ -61,26 +61,114 @@ suite('CalcLength', function() {
   });
 
   test('Multiplication of a CalcLength length produces a new CalcLength object', function() {
-    var calcLength = new CalcLength({px: 10, em: 3.2})
-    var multipliedCalcLength = calcLength.multiply(4);
-    assert.strictEqual(multipliedCalcLength.cssString, 'calc(40px+12.8em)');
+    var calcLength = new CalcLength({px: 10, em: 3.2});
+    var result = calcLength.multiply(4);
+    assert.strictEqual(result.cssString, 'calc(40px+12.8em)');
   });
 
   test('Multiplication of a decimal number produces expected result', function() {
-    var calcLength = new CalcLength({px: 10, em: 3.2})
-    var multipliedCalcLength = calcLength.multiply(0.5);
-    assert.strictEqual(multipliedCalcLength.cssString, 'calc(5px+1.6em)');
+    var calcLength = new CalcLength({px: 10, em: 3.2});
+    var result = calcLength.multiply(0.5);
+    assert.strictEqual(result.cssString, 'calc(5px+1.6em)');
   });
 
   test('Division of a CalcLength length produces a new CalcLength object', function() {
-    var calcLength = new CalcLength({px: 10, em: 4.0})
-    var dividedCalcLength = calcLength.divide(4);
-    assert.strictEqual(dividedCalcLength.cssString, 'calc(2.5px+1em)');
+    var calcLength = new CalcLength({px: 10, em: 4.0});
+    var result = calcLength.divide(4);
+    assert.strictEqual(result.cssString, 'calc(2.5px+1em)');
   });
 
   test('Division of a decimal number produces expected result', function() {
-    var calcLength = new CalcLength({px: 25, em: 3.2})
-    var dividedCalcLength = calcLength.divide(2.5);
-    assert.strictEqual(dividedCalcLength.cssString, 'calc(10px+1.28em)');
+    var calcLength = new CalcLength({px: 25, em: 3.2});
+    var result = calcLength.divide(2.5);
+    assert.strictEqual(result.cssString, 'calc(10px+1.28em)');
+  });
+
+  test('Adding two CalcLengths returns a new CalcLength with expected values in each value type', function() {
+    var calcLength1 = new CalcLength({px: 15, em: 6});
+    var calcLength2 = new CalcLength({px: 10, em: 3});
+    var result = calcLength1.add(calcLength2);
+    var expectedResult = new CalcLength({px: 25, em: 9});
+
+    assert.instanceOf(result, CalcLength, 'two added CalcLength of same type should be an instance of CalcLength');
+    assert.isTrue(expectedResult.equals(result));
+  });
+
+  test('All values in added CalcLengths appear in the output', function() {
+    var calcLength1 = new CalcLength({px: 15, em: 6, percent: 5});
+    var calcLength2 = new CalcLength({px: 10, em: 3, ex: 6});
+    var result = calcLength1.add(calcLength2);
+    var expectedResult = new CalcLength({px: 25, em: 9, percent: 5, ex: 6});
+
+    assert.instanceOf(result, CalcLength, 'two added CalcLength of same type should be an instance of CalcLength');
```

Comment:
And actually we don't really need all of these instanceOf asserts for add etc. They are repetitive, but also LengthValue.equals checks types. You can leave in one for each method (one for add, one for subtract, ect.) for the sake of explicitness if you want. The rest can go.